### PR TITLE
[IMP] auth_oauth: update help for GMail login

### PR DESCRIPTION
**Pull request by hmo-odoo**
Task: https://www.odoo.com/web#id=12181&view_type=form&model=project.task&menu_id=3942&action=327

Pad: https://pad.odoo.com/p/r.RQRAT2jnFWkIHyra

### CODE REVIEW

File: addons/auth_oauth/res_config.xml [hunk 1]
```diff
@@ -15,15 +15,24 @@
...
-                                    To setup the signin process with Google, first you have to perform the following steps:<br/>
-                                    <br/>
-                                    - Go to the <a href="https://code.google.com/apis/console/">Google APIs console</a><br/>
-                                    - Ceate a new project<br/>
-                                    - Go to Api Access<br/>
-                                    - Create an oauth client_id<br/>
-                                    - Edit settings and set both Authorized Redirect URIs and Authorized JavaScript Origins to your hostname.<br/>
-                                    <br/>
-                                    Now copy paste the client_id here: <field name="auth_oauth_google_client_id" class="oe_inline" placeholder="e.g. 1234-xyz.apps.googleusercontent.com"/>
+                                    <p style="font-size:13px;color: rgb(76, 76, 76);">To setup the signin process with Google, first you have to perform the following steps:<br/>
```
Comment (hmo-odoo) [inline addons/auth_oauth/res_config.xml]: Please do not use inline / custom css .

why use <br/> ? can possible use <div> tag ?

File: addons/auth_oauth/res_config.xml [hunk 2]
```diff
@@ -8,29 +8,48 @@
...
+                                        </li>
+                                        <li class="mb8">In the menu on left side, select the sub menu "Credentials" (from menu APIs and auth) then select "oAuth consent screen".</li>
+                                        <div class="clearfix"></div>
+                                        <img class="mb8" src='/auth_oauth/static/src/img/steps_3.png'/>
+                                        <li class="mb8">Fill your address, email and the product name (for example odoo) and then save.
+                                        </li>
+                                        <img class="mb8" src='/auth_oauth/static/src/img/steps_4.png'/>
+                                        <li class="mb8">In the menu on left side, select the sub menu Credentials (from menu APIs and auth) and click on Credentials. Then click on "add Credentials" select second option(oAuth 2.0 clinet ID).</li>
+                                        <img class="mb8" src='/auth_oauth/static/src/img/steps_5.png'/>
+                                        <div class="clearfix"></div>
```
Comment (rga-odoo) [inline addons/auth_oauth/res_config.xml]: why after each li need `clearfix` ?

File: addons/base_setup/res_config_view.xml [hunk 3]
```diff
@@ -53,9 +53,11 @@
                     <group name="authentication">
                         <label for="id" string="Authentication"/>
                         <div>
+                            <div>
                             <field name="module_auth_oauth" class="oe_inline"/>
                             <label for="module_auth_oauth"/>
                         </div>
+                    </div>
```
Comment (rga-odoo) [inline addons/base_setup/res_config_view.xml]: element indentation open/close miss match

File: addons/auth_oauth/res_config.py [hunk 4]
```diff
@@ -9,9 +10,15 @@
...
 
+    def _default_server_uri(self):
+        server_uri_gmail = self.pool.get('ir.config_parameter').get_param(self._cr, SUPERUSER_ID, 'web.base.url')
+        server_uri = "%s/auth_oauth/signin" % (server_uri_gmail)
+        return server_uri
+
     _columns = {
         'auth_oauth_google_enabled' : fields.boolean('Allow users to sign in with Google'),
         'auth_oauth_google_client_id' : fields.char('Client ID'),
+        'server_uri_gmail': fields.char(default=_default_server_uri, store=False, string='Server uri'),
```
Comment (rim-odoo) [inline addons/auth_oauth/res_config.py]: `store=False` is the default...

File: addons/auth_oauth/static/src/less/auth_oauth.less [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.gmail_auth_steps {
+    margin-left: 50px;
+}
```
Comment (rim-odoo) [inline addons/auth_oauth/static/src/less/auth_oauth.less]: Remove this, it is overkill (and anyway, the page looks clean without)

File: addons/auth_oauth/views/auth_oauth_login_templates.xml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="assets_backend" name="auth oauth" inherit_id="web.assets_backend">
+        <xpath expr="." position="inside">
+            <link rel="stylesheet" href="/auth_oauth/static/src/less/auth_oauth.less"/>
+        </xpath>
+    </template>
+</odoo>
```
Comment (rim-odoo) [inline addons/auth_oauth/views/auth_oauth_login_templates.xml]: remove this

File: addons/auth_oauth/res_config.xml [hunk 7]
```diff
@@ -14,23 +14,73 @@
...
+                                <ol>
+                                    <li>
+                                        Connect on your google account and go to the
+                                        <a href="https://console.developers.google.com/">https://console.developers.google.com/</a>
+                                    </li>
+                                    <li>
+                                        Click on <b>"Create Project"</b> and enter the project name and other details.
+                                    </li>
+                                        <p>
+                                            <img class="gmail_auth_steps mt8" src='/auth_oauth/static/src/img/steps_1.png'/>
```
Comment (rim-odoo) [inline addons/auth_oauth/res_config.xml]: remove all gmail_auth_steps

File: addons/auth_oauth/res_config.py [hunk 8]
```diff
@@ -9,9 +10,15 @@
 class base_config_settings(osv.TransientModel):
     _inherit = 'base.config.settings'
 
+    def _default_server_uri(self):
```
Comment (rim-odoo) [inline addons/auth_oauth/res_config.py]: you are making a new api method in an old api class...

File: addons/auth_oauth/res_config.py [hunk 9]
```diff
@@ -9,9 +10,15 @@
...
+        server_uri_gmail = self.pool.get('ir.config_parameter').get_param(self._cr, SUPERUSER_ID, 'web.base.url')
```
Comment (rim-odoo) [inline addons/auth_oauth/res_config.py]: self.pool is old api, self._cr is new api, SUPERUSER_ID is old api... make a choice and stick it!

File: addons/auth_oauth/res_config.xml [hunk 10]
```diff
@@ -14,23 +14,73 @@
...
+                                    </p>
+                                    <img class="gmail_auth_steps" src='/auth_oauth/static/src/img/steps_6.png'/>
+                                    <li class="mt8">
+                                        Check that the application type is set on <b>"Web Application"</b>. Now configure the allowed pages on which you will be redirected.
+                                    </li>
+                                    <li>
+                                        To do it, you need to complete the field <b>"Authorized redirect URIs"</b> and set as value (your own domain followed by<b> '/auth_oauth/signin' </b>).Then click on <b>"Create"</b>.
+                                    </li>
+                                    <div>
+                                        <b>===> <field name="server_uri_gmail" readonly="1" class="oe_inline"/></b>
```
Comment (rim-odoo) [inline addons/auth_oauth/res_config.xml]: Are you really sure this line was requested in @bst-odoo 's original text? Please stick to the spec...

Comment (rka-odoo) [inline addons/auth_oauth/res_config.xml]: @rim-odoo I have put that as it needed to guide the user to enter it in 'Authorized redirect url' as it's also for google calendar tutorial.

File: addons/auth_oauth/res_config.xml [hunk 11]
```diff
@@ -6,31 +6,86 @@
...
+                                    </li>
+                                    <p>
+                                        <img src='/auth_oauth/static/src/img/steps_5.png'/>
+                                    </p>
+                                    <img src='/auth_oauth/static/src/img/steps_6.png'/>
+                                    <li class="mt8">
+                                        Check that the application type is set on <b>"Web Application"</b>. Now configure the allowed pages on which you will be redirected.
+                                    </li>
+                                    <li>
+                                    To do it, you need to complete the field <b>"Authorized redirect URIs."</b>Copy paste the following link in the box:<b>===><field name="server_uri_gmail" readonly="1" class="oe_inline"/></b>
```
Comment (bst-odoo) [inline addons/auth_oauth/res_config.xml]: remove the arrow : ===>

File: addons/auth_oauth/res_config.py [hunk 12]
```diff
@@ -9,9 +9,18 @@
...
+        return "%s/auth_oauth/signin" % (self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context))
+
+    def _get_server_uri(self, cr, uid, ids, name, arg, context=None):
+        return dict.fromkeys(ids, self.get_uri(cr, uid, context))
+
     _columns = {
         'auth_oauth_google_enabled' : fields.boolean('Allow users to sign in with Google'),
         'auth_oauth_google_client_id' : fields.char('Client ID'),
+        'server_uri_gmail': fields.function(_get_server_uri, type='char', string='Server uri'),
+        'module_auth_oauth': fields.boolean('Use external authentication providers (OAuth)'),
```
Comment (rga-odoo) [inline addons/auth_oauth/res_config.py]: why redefine `module_auth_oauth`?

Comment (rka-odoo) [inline addons/auth_oauth/res_config.py]: @rga-odoo it's become very complex if we inherit view an add xpath, for the simplicity i redefine 'module_auth_oauth' field.

Comment (rim-odoo) [inline addons/auth_oauth/res_config.py]: lol

File: addons/auth_oauth/res_config.py [hunk 13]
```diff
@@ -9,9 +9,17 @@
...
+    def get_uri(self, cr, uid, context=None):
+        return "%s/auth_oauth/signin" % (self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context))
+
+    def _get_server_uri(self, cr, uid, ids, name, arg, context=None):
+        return dict.fromkeys(ids, self.get_uri(cr, uid, context))
+
     _columns = {
         'auth_oauth_google_enabled' : fields.boolean('Allow users to sign in with Google'),
         'auth_oauth_google_client_id' : fields.char('Client ID'),
+        'server_uri_gmail': fields.function(_get_server_uri, type='char', string='Server uri'),
```
Comment (hmo-odoo) [inline addons/auth_oauth/res_config.py]: rename with server_uri_google

File: addons/auth_oauth/res_config.xml [hunk 14]
```diff
@@ -6,31 +6,86 @@
...
+                                    </li>
+                                    <p>
+                                        <img src='/auth_oauth/static/src/img/steps_5.png'/>
+                                    </p>
+                                    <img src='/auth_oauth/static/src/img/steps_6.png'/>
+                                    <li class="mt8">
+                                        Check that the application type is set on <b>"Web Application"</b>. Now configure the allowed pages on which you will be redirected.
+                                    </li>
+                                    <li>
+                                    To do it, you need to complete the field <b>"Authorized redirect URIs."</b>Copy paste the following link in the box:<b><field name="server_uri_gmail" readonly="1" class="oe_inline"/></b>
```
Comment (hmo-odoo) [inline addons/auth_oauth/res_config.xml]: do not use oe_inline class.

also remove readonly="1"

File: addons/auth_oauth/res_config.py (hunk 13 above)
Comment (rim-odoo) [inline addons/auth_oauth/res_config.py]: then put an oldname attribute

**Comment by hmo-odoo** [comment]
use o_row in div instead of oe_inline class

File: addons/base_setup/res_config_view.xml [hunk 15]
```diff
@@ -44,8 +44,10 @@
                     <group name="authentication">
                         <label for="id" string="Authentication"/>
                         <div>
-                            <field name="module_auth_oauth" class="oe_inline"/>
-                            <label for="module_auth_oauth"/>
+                            <div>
+                                <field name="module_auth_oauth" class="oe_inline"/>
+                                <label for="module_auth_oauth"/>
```
Comment (msh-odoo) [inline addons/base_setup/res_config_view.xml]: remove oe_inline from here also(use o_row)

File: addons/auth_oauth/res_config.py [hunk 16]
```diff
@@ -22,9 +31,11 @@ def default_get(self, cr, uid, fields, context=None):
     def get_oauth_providers(self, cr, uid, fields, context=None):
         google_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'auth_oauth', 'provider_google')[1]
         rg = self.pool.get('auth.oauth.provider').read(cr, uid, [google_id], ['enabled','client_id'], context=context)
+        uri = self.get_uri(cr, uid, context)
         return {
             'auth_oauth_google_enabled': rg[0]['enabled'],
             'auth_oauth_google_client_id': rg[0]['client_id'],
+            'server_uri_google': uri
```
Comment (msh-odoo) [inline addons/auth_oauth/res_config.py]: ``` 'server_uri_google': self.get_uri(cr, uid, context) ```

File: addons/auth_oauth/res_config.py [hunk 17]
```diff
@@ -9,9 +9,17 @@
...
+    def get_uri(self, cr, uid, context=None):
+        return "%s/auth_oauth/signin" % (self.pool['ir.config_parameter'].get_param(cr, uid, 'web.base.url', context=context))
+
+    def _get_server_uri(self, cr, uid, ids, name, arg, context=None):
+        return dict.fromkeys(ids, self.get_uri(cr, uid, context))
+
     _columns = {
         'auth_oauth_google_enabled' : fields.boolean('Allow users to sign in with Google'),
         'auth_oauth_google_client_id' : fields.char('Client ID'),
+        'server_uri_google': fields.function(_get_server_uri, type='char', string='Server uri', oldname='server_uri_gmail'),
```
Comment (tivisse) [inline addons/auth_oauth/res_config.py]: Oldname that doesn't exists ?

**Comment by tivisse** [comment]
You could optimize the pictures with pngquant

**Pull request merged**